src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/X509Pal.cs

```diff
@@ -130,6 +140,41 @@ private static RSA DecodeRsaPublicKey(byte[] encodedKeyValue)
                     throw;
                 }
             }
+
+            private static DSA DecodeDsaPublicKey(byte[] encodedKeyValue, byte[] encodedParameters)
+            {
+                SubjectPublicKeyInfoAsn spki = new SubjectPublicKeyInfoAsn
+                {
+                    Algorithm = new AlgorithmIdentifierAsn { Algorithm = Oids.Dsa, Parameters = encodedParameters },
+                    SubjectPublicKey = encodedKeyValue,
+                };
+
+                AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+                spki.Encode(writer);
+
+                byte[] rented = CryptoPool.Rent(writer.GetEncodedLength());
+
+                if (!writer.TryEncode(rented, out int written))
+                {
+                    Debug.Fail("TryEncode failed with a pre-allocated buffer");
+                    throw new InvalidOperationException();
+                }
```

Comment:
This boiler-plate can be written easier:

```suggestion
                int written = writer.Encode(rented);
```

It's just an "older code vs new API" artifact that there are so many blocks like this through the codebase.